src/java.base/share/classes/java/util/concurrent/StructuredTaskScopeImpl.java

```diff
@@ -64,6 +64,7 @@ final class StructuredTaskScopeImpl<T, R> implements StructuredTaskScope<T, R> {
     private StructuredTaskScopeImpl(Joiner<? super T, ? extends R> joiner,
                                     ThreadFactory threadFactory,
                                     String name) {
+        this.state = ST_NEW;
         this.joiner = joiner;
         this.threadFactory = threadFactory;
         this.flock = ThreadFlock.open((name != null) ? name : Objects.toIdentityString(this));
```

Comment:
Unnecessary but okay. Do you mind moving it to the end of constructor as the fields are initialized in order declaration order in this constructor.

Reply:
Yeah, explicit use of `ST_NEW` avoids future accidents, e.g. if we ever introduce a new state that would have to shuffle states around. Moved to the end of constructor now.